src/V3Const.cpp

```diff
@@ -2447,6 +2447,7 @@ class ConstVisitor final : public AstNVisitor {
     TREEOPV("AstRedAnd{$lhsp.castExtend, $lhsp->width() > VN_CAST($lhsp,,Extend)->lhsp()->width()}", "replaceZero(nodep)");  // &{0,...} => 0  Prevents compiler limited range error
     TREEOPV("AstRedOr {$lhsp.castExtend}",      "AstRedOr {$lhsp->castExtend()->lhsp()}");
     TREEOPV("AstRedXor{$lhsp.castExtend}",      "AstRedXor{$lhsp->castExtend()->lhsp()}");
+    TREEOPC("AstRedXor{$lhsp.castXor, VN_IS(VN_CAST($lhsp,,Xor)->lhsp(),,Const)}", "AstXor{AstRedXor{$lhsp->castXor()->lhsp()}, AstRedXor{$lhsp->castXor()->rhsp()}}");  // ^(const ^ a) => (^const)^(^a)
```

Comment:
I suspect this can be a normal "TREEOP" that is do it anytime.

Reply:
I think it would work with TREEOP too.
I use TREEOPC because that is enough for #2751 
I thought TREEOPC is cheaper than TREEOP because it is called only in C++ mode.

I am fine with changing to TREEOP if it would show better result.

Reply:
This will run quick, so fine as TREEOP and might optimize better.